base/intfuncs.jl

```diff
@@ -379,8 +379,22 @@ true
 
 julia> ispow2(5)
 false
+
+julia> ispow2(4.5)
+false
+
+julia> ispow2(0.25)
+true
+
+julia> ispow2(1//8)
+true
 ```
+
+!!! compat "Julia 1.6"
+    Support for non-`Integer` arguments was added in Julia 1.6.
 """
+ispow2(x::Number) = isreal(x) && ispow2(real(x))
```

Comment:
Maybe this should be restricted to `Complex`, since there isn't a fallback covering all Real values.

Reply:
I feel like there should be a way to throw a `MethodError` for common patterns like this, rather than `StackOverflowError`.  I've bumped into similar issues many times.

Could we implement something like
```jl
@norecurse ispow2(x::Number) = isreal(x) && ispow2(real(x))
```

Reply:
(For example, this `Number` method would work perfectly well for quaternion types.  It seems a shame to lose genericity due to dispatch limitations.)

Reply:
I agree.

Reply:
With @stevengj's approach ^, the work of making types consistent inside and collaboratively correct outside is a touch simpler. 
```
@norecurse fn(x::Real) = fn(float(x))
```
_dispatch resolves, giving attentive safety_